mne_bids/path.py

```diff
@@ -394,17 +396,56 @@ def update(self, check=None, **entities):
         if self.datatype is None and \
                 self.suffix in SUFFIX_TO_DATATYPE:
             self.datatype = SUFFIX_TO_DATATYPE[self.suffix]
-        # set datatype based on suffix if calling update
-        elif self.suffix in SUFFIX_TO_DATATYPE and \
-                'datatype' not in entities:
-            self.datatype = SUFFIX_TO_DATATYPE[self.suffix]
+        # # set datatype based on suffix if calling update
+        # elif self.suffix in SUFFIX_TO_DATATYPE and \
+        #         'datatype' not in kwargs:
+        #     self.datatype = SUFFIX_TO_DATATYPE[self.suffix]
```

Comment:
I like this, generally. (the removal of this heuristic)

But I believe there were reasons why we added this in the first place, and there may be other places where we're doing the same. @adam2392, could you share your thoughts please?